### ISSUE TITLE
JPEG image renders with a horizontal line across middle of image when Exif Orientation tag changes

### ISSUE THREAD

**Issue by chunyang** [Beep6581/RawTherapee#6924]
## Short description

When opening a JPEG image, depending on the Exif Orientation tag of the image, the RawTherapee UI may render a horizontal line across the middle of the image as in the following example:

![rt-ui](https://github.com/Beep6581/RawTherapee/assets/454684/ee421e26-1c1f-494d-9ed4-d798a0b9950d)

When the image is saved from RawTherapee, the horizontal line is also saved to the resulting file.

## Steps to reproduce

1. Open the following image in RawTherapee: ![000154400034](https://github.com/Beep6581/RawTherapee/assets/454684/4911bffb-0934-4027-a634-8f2aaa6a1432)
2. Observe the horizontal line in the middle of the image in the UI.

## Expected behavior

The UI does not render a horizontal line in the middle of the image.

### RawTherapee version
```
Version: 5.9
Branch: releases
Commit: 9b8583988
Commit date: 2022-11-27
Compiler: cc 12.2.0
Processor: x86_64
System: Linux
Bit depth: 64 bits
Gtkmm: V3.24.7
Lensfun: V0.3.3.0
Build type: Release
Build flags: -march=x86-64 -mtune=generic -O2 -pipe -fno-plt -fexceptions         -Wp,-D_FORTIFY_SOURCE=2 -Wformat -Werror=format-security         -fstack-clash-protection -fcf-protection -Wp,-D_GLIBCXX_ASSERTIONS -flto=auto -std=c++11 -ffp-contract=off  -Werror=unused-label -Werror=delete-incomplete -fno-math-errno -Wno-attributes -Wall -Wuninitialized -Wcast-qual -Wno-deprecated-declarations -Wno-unused-result -Wunused-macros -fopenmp -Werror=unknown-pragmas -O3 -DNDEBUG -ftree-vectorize
Link flags: -Wl,-O1,--sort-common,--as-needed,-z,relro,-z,now -flto=auto
OpenMP support: ON
MMAP support: ON
Build OS: 
Build date:  UTC
Build epoch: 
Build UUID: 
```

### OS version
Linux guava 6.6.8-arch1-1 #1 SMP PREEMPT_DYNAMIC Thu, 21 Dec 2023 19:01:01 +0000 x86_64 GNU/Linux

### Other information
When the Exif Orientation is changed (from 8 to 3), RawTherapee loads the image correctly.
![rt-ui-ok](https://github.com/Beep6581/RawTherapee/assets/454684/fe1c4d22-118d-401e-ae5f-b37be53bfea1)
![000154400034](https://github.com/Beep6581/RawTherapee/assets/454684/d7eb993c-97f4-413b-ac34-5a162aabc241)

```
% exiv2 -pv 000154400034.jpg | grep Image.*Orientation
0x0112 Image        Orientation                 Short       1  8  # <- Problematic image has this orientation
0x0112 Image        Orientation                 Short       1  3  # <- Working as expected image has this orientation
```

**Comment by Lawrence37** [comment]
Thanks for the bug report. I have proposed a fix in the linked pull request.

**Comment by chunyang** [comment]
Thanks for taking a look and making the fix!

**Comment by chunyang** [comment]
Closed by #6926
